Fix array_walk() call on PHP 8

derrabus opened the pull request:
Running version 6.7 of this library on PHP 8 will result in a warning being triggered:

```
Argument #2 ($key) must be passed by reference, value given
```

The fix is fairly trivial because the affected variable is not used and can be removed.

This issue has already been reported as #1085 fixed on higher branches. However the project I'm working with at the moment is currently unable to upgrade to Elasticsearch 7. And as far as I can tell, this issue seems to be our only blocker for upgrading to PHP 8.

If the 6.7.x branch is closed for bugfixes, please close this PR.

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

alexander-schranz commented:
The ones which did subscribe this PR. Elasticsearch did release a PHP 8 supported Version with 7.11: https://github.com/elastic/elasticsearch-php/releases/tag/v7.11.0 not sure if they will support PHP 8 with older version maybe @ezimuel can make a statement about it.

ezimuel commented:
The original plan was to support PHP 8 only for `elasticsearch-php` 7.x (I just released it yesterday with [7.11.0](https://github.com/elastic/elasticsearch-php/releases/tag/v7.11.0)). Considering that elasticsearch 6.x is still [quite popular](https://packagist.org/packages/elasticsearch/elasticsearch/stats#major/6) I can consder to add the support of PHP 8 al well.

@derrabus can you send the PR to **6.x**? Thanks!

derrabus commented:
Sure, no problem. I've changed the target branch and rebased the changes.

If you need more help building a PHP8-compatible 6.x release, let me know.

derrabus commented:
The Travis failures are unrelated to my changes. I'm trying to fix them in #1105.

derrabus commented:
I think we can close the PR because the problem has been addressed in b316710c26e7f96f949f8a0c54088e6df23ead69. I've switched our project to the 6.8.x branch of this library and it works like a charm!

Thank you very much, @ezimuel!
